bigtable/google/cloud/bigtable/row_data.py

```diff
@@ -434,12 +438,10 @@ def cancel(self):
 
     def _create_retry_request(self):
         """Helper for :meth:`read_rows`."""
-        row_range = self.request.rows.row_ranges.pop()
-        range_kwargs = {}
-        # start AFTER the row_key of the last successfully read row
-        range_kwargs['start_key_open'] = self.last_scanned_row_key
-        range_kwargs['end_key_open'] = row_range.end_key_open
-        self.request.rows.row_ranges.add(**range_kwargs)
+        req_manager = _ReadRowsRequestManager(self.request,
+                                              self.last_scanned_row_key,
+                                              self._counter)
+        self.request = req_manager.build_updated_request()
```

Comment:
Would it make sense to convert `_ReadRowsRequestManager. build_updated_request` to a new method?

Reply:
@sduskis : I keep retry logic separate from YieldRowsData for more clarity. Yes, we can have a build_updated_request()  method in the YieldRowsData class as well. In that case, there wont be a need of _ReadRowsRequestManager() class.